feature/system/tests/system_base_test/g_protocol_test.go

```diff
@@ -0,0 +1,112 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package system_base_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/openconfig/ondatra"
+	"google.golang.org/grpc"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	spb "github.com/openconfig/gnoi/system"
+	authzpb "github.com/openconfig/gnsi/authz"
+	gribipb "github.com/openconfig/gribi/v1/proto/service"
+	p4rtpb "github.com/p4lang/p4runtime/go/p4/v1"
+)
+
+// TestGNMIClient validates that the DUT listens on standard gNMI Port.
+func TestGNMIClient(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	addr := fmt.Sprintf("%s:9339", dut.Name())
+	t.Logf("gNMI standard port test: %q", addr)
+	conn, err := grpc.Dial(addr, grpc.WithBlock())
+	if err != nil {
+		t.Fatalf("grpc.Dial failed to: %q", addr)
+	}
+	c := gpb.NewGNMIClient(conn)
+	_, err = c.Get(context.Background(), &gpb.GetRequest{})
+	if err != nil {
+		t.Fatalf("gnmi.Get failed: %v", err)
+	}
+}
+
+// TestGNOIClient validates that the DUT listens on standard gNMI Port.
+func TestGNOIClient(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	addr := fmt.Sprintf("%s:9339", dut.Name())
+	t.Logf("gNOI standard port test: %q", addr)
+	conn, err := grpc.Dial(addr, grpc.WithBlock())
+	if err != nil {
+		t.Fatalf("grpc.Dial failed to: %q", addr)
+	}
+	c := spb.NewSystemClient(conn)
+	_, err = c.Time(context.Background(), &spb.TimeRequest{})
```

Comment:
We should ensure that the featureprofile here requires us to have `gNOI.Time` implemented. I don't recall whether it's been spelled out as an explicit requirement for any specific test thus far.

Reply:
it is not required :P

Reply:
Confused! What happens if the server returns `Unimplemented` here -- this test fails at that point, right?